src/Grafana.OpenTelemetry.Base/MeterProviderBuilderExtensions.cs

```diff
@@ -67,5 +68,22 @@ internal static MeterProviderBuilder AddInstrumentations(this MeterProviderBuild
 
             return builder;
         }
+
+        internal static MeterProviderBuilder DropUnusedMetrics(this MeterProviderBuilder builder, GrafanaOpenTelemetrySettings settings)
+        {
+            if (settings.DataSaverEnabled)
+            {
+                return builder;
+            }
+
+            var allow = new MetricStreamConfiguration();
+
+            return builder.AddView("application", allow)
+                .AddView("process.cpu.time", allow)
+                .AddView("memory.usage", allow)
+                .AddView("process.threads", allow)
+                .AddView("process.runtime.dotnet.gc.objects.size", allow)
+                .AddView("*", MetricStreamConfiguration.Drop);
```

Comment:
In my opinion it's problematic to drop _everything_ besides those four metrics. We can drop metrics added via instrumentation libraries that our distro pulls in, but we should avoid dropping custom metrics.

I see two possible alternative solutions.

1. We maintain a customizable lists of metrics that are dropped (the inverse of the proposed approach). That's more effort, as we need to maintain a possible large list of metrics, but it's the most flexible approach.

2. We only drop metrics from instrumentation libraries we know. Like this (untested):
```csharp
.AddView((instrument) =>
    {
        if (knownMeterNames.Contains(instrument.Meter.Name))
        {
            return MetricStreamConfiguration.Drop;
        }

        return allow;
    })
```

In general, I'd be more cautious in what we drop. I'm in favor of a solution based on (1), where we maintain a list of metrics that we think are of no use to most users. For example, I wouldn't drop metrics for HTTP server request duration, as those can be very valuable for users. However, we could safely ignore ASP.NET Core specific metrics like those listed [here](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/dotnet/dotnet-aspnetcore-metrics.md).